src/libYARP_sig/src/Image.cpp

```diff
@@ -604,12 +606,35 @@ int ImageStorage::_pad_bytes (int linesize, int align) const
     return yarp::sig::PAD_BYTES (linesize, align);
 }
 
-
-
-
-
-
-
+const std::map<YarpVocabPixelTypesEnum, unsigned int> Image::pixelCode2Size = {
+    {VOCAB_PIXEL_INVALID,               0 },
+    {VOCAB_PIXEL_MONO,                  sizeof(yarp::sig::PixelMono)},
+    {VOCAB_PIXEL_MONO16,                sizeof(yarp::sig::PixelMono16)},
+    {VOCAB_PIXEL_RGB,                   sizeof(yarp::sig::PixelRgb)},
+    {VOCAB_PIXEL_RGBA,                  sizeof(yarp::sig::PixelRgba)},
+    {VOCAB_PIXEL_BGRA,                  sizeof(yarp::sig::PixelBgra)},
+    {VOCAB_PIXEL_INT,                   sizeof(yarp::sig::PixelInt)},
+    {VOCAB_PIXEL_HSV,                   sizeof(yarp::sig::PixelHsv)},
+    {VOCAB_PIXEL_BGR,                   sizeof(yarp::sig::PixelBgr)},
+    {VOCAB_PIXEL_MONO_SIGNED,           sizeof(yarp::sig::PixelMonoSigned)},
+    {VOCAB_PIXEL_RGB_SIGNED,            sizeof(yarp::sig::PixelRgbSigned)},
+    {VOCAB_PIXEL_RGB_INT,               sizeof(yarp::sig::PixelRgbInt)},
+    {VOCAB_PIXEL_MONO_FLOAT,            sizeof(yarp::sig::PixelFloat)},
+    {VOCAB_PIXEL_RGB_FLOAT,             sizeof(yarp::sig::PixelRgbFloat)},
+    {VOCAB_PIXEL_HSV_FLOAT,             sizeof(yarp::sig::PixelHsvFloat)},
+    {VOCAB_PIXEL_ENCODING_BAYER_GRBG8,  8 },
+    {VOCAB_PIXEL_ENCODING_BAYER_GRBG16, 16},
+    {VOCAB_PIXEL_ENCODING_BAYER_BGGR8,  8 },
+    {VOCAB_PIXEL_ENCODING_BAYER_BGGR16, 16},
+    {VOCAB_PIXEL_ENCODING_BAYER_GBRG8,  8 },
+    {VOCAB_PIXEL_ENCODING_BAYER_GBRG16, 16},
+    {VOCAB_PIXEL_ENCODING_BAYER_RGGB8,  8 },
+    {VOCAB_PIXEL_ENCODING_BAYER_RGGB16, 16},
+    {VOCAB_PIXEL_YUV_420,               3 },
+    {VOCAB_PIXEL_YUV_444,               3 },
+    {VOCAB_PIXEL_YUV_422,               3 },
+    {VOCAB_PIXEL_YUV_411,               3 },
```

Comment:
The sizes of pixels for bayer and yuv does not seem right to me. Where did you get those values?
I think they should be:
{VOCAB_PIXEL_ENCODING_BAYER_GRBG8,       1 },   // Values are in bytes, 8/16 were bits instead
{VOCAB_PIXEL_ENCODING_BAYER_GRBG16,     2},
{VOCAB_PIXEL_ENCODING_BAYER_BGGR8,       1 },
{VOCAB_PIXEL_ENCODING_BAYER_BGGR16,     2},
{VOCAB_PIXEL_ENCODING_BAYER_GBRG8,       1 },
{VOCAB_PIXEL_ENCODING_BAYER_GBRG16,     2},
{VOCAB_PIXEL_ENCODING_BAYER_RGGB8,       1 },
{VOCAB_PIXEL_ENCODING_BAYER_RGGB16,     2},

for yuv is a bit more tricky. The size should be:

{VOCAB_PIXEL_YUV_444,               3 },
{VOCAB_PIXEL_YUV_422,               2 },
{VOCAB_PIXEL_YUV_411,               1.5 },  --> how can we handle this? increase it to 2?
{VOCAB_PIXEL_YUV_420,               1.5},   --> how can we handle this? increase it to 2?  Is it ever been used?

Reply:
ok.. i fixed up the bayer ones.. it was just a slip. 

For the YUV ones i don't get the code..  if the numbers are bit used we would have 
VOCAB_PIXEL_YUV_444, 1.5
VOCAB_PIXEL_YUV_422, 1
VOCAB_PIXEL_YUV_411, 0.75
VOCAB_PIXEL_YUV_420, 0.75

Reply:
Actually the code does not refers to bits, but to subsampling.
Here is a short video explaining them, for reminder
http://www.personal-view.com/talks/discussion/6144/easy-video-explaining-444-422-and-420-for-noobs/p1